addon/adapters/application.js

```diff
@@ -142,18 +141,25 @@ export default Ember.Object.extend(FetchMixin, Evented, {
     @return {Promise}
   */
   createResource(resource) {
-    let url = this.get('url');
-    const json = this.serializer.serialize(resource);
-    return this.fetch(url, {
+    return this.fetch(this.get('url'), {
       method: 'POST',
-      body: JSON.stringify(json)
+      body: JSON.stringify(this.serializer.serialize(resource))
     }).then(function(resp) {
       if (resource.toString().match('JSONAPIResource') === null) {
         return resp;
```

Comment:
I'm not sure what you're referring to here. I'm not clear on what is happening after `then()` here. What is the purpose of `if (resource.toString().match('JSONAPIResource') === null) {` ?

I think I interpreted that as checking if the resource was new or somehow was expected to be the same as the response, but I don't see how I got that from this line.

I'm suddenly quite confused about this method.
- `createResource` should only be called on new resources, but there is nothing preventing this. Should we? (I don't think so, since we don't really enforce "newness")
- When successful, we should update the resource with properties gotten from response (as happens from line #151). Always. No exception.
- Return original resource.
- That's it. But that leaves some questions:
  - What is `return resp` doing? I think the tests show that `resp` is already a `Resource` instance, so `fetch` might already update the resource? Is it?!
  - And of course, what is that `if (resource.toString().match())` line doing?
  - Who handles/catches the error? The user I guess (should be ok)
  - If fetch is successful (user won't catch that error) are there situation we should handle/check?
    - I can imagine: `createResource` called with resource that already has an `id`, but server ignores the id on POST (since that's not supposed to be used to update a resource), creates a new resource with the given properties, and returns them all with a different `id`. If we then simply update the `id` on the given `resource` we dun goofed.
     - I imagine we handle this like so: 
         - `isNew`? Error about attempting to create an existing resource. No POST for you.
         - `isNew`? We POST since you asked nicely, but we error on receiving a mismatching `id`.

I'll dive into what `fetch` is doing exactly and where resources are instantiated exactly.

Also, I think I dun goofed myself by doing `this.get('url')` here, that should of course be `this.fetchUrl()`. Though that's what it did originally. But that's wrong, right? (needs `fetchUrl`)